### ISSUE TITLE
1) Remote Start fails; 2) Drops authentication

### ISSUE THREAD

**Issue by ttwajsb** [itsMeDavidV/Watchla-Support#351]
1) Remote Start in Tesla app works fine, but on Watch Watchla says “Remote start enabled” yet it's not.
2) Every couple days must go thru Watch reauthentication process. Watchla repeatedly forgets it's been authorized, even though within Tesla app I can see it's been approved since 3/4/24. 

None of these conditions apply:
![image](https://github.com/itsMeDavidV/Watchla-Support/assets/9358980/a668bc3c-8517-46d4-89aa-b9700d8d3e8e)

This is simply untrue:
![image](https://github.com/itsMeDavidV/Watchla-Support/assets/9358980/e5326808-3738-4e34-b3e9-567d381b2f1f)

Repeatedly encountering this:
![image](https://github.com/itsMeDavidV/Watchla-Support/assets/9358980/61968b0f-4dc6-49af-b37d-b870d582bf0e)

**Comment by itsMeDavidV** [comment]
@ttwajsb yeah you'll need to re-authenticate the watch app it looks like. The good news is that we are testing a an auth management change that will prevent this in the future. Are you still in beta?

**Comment by ttwajsb** [comment]
Beta Watchla, or FSD Beta? I'm now on 12.3.3 “Supervised”. 

I've authenticated the Watch and finally, after months without it, have Remote Start working. Yippee!

Only issue now is how long it takes for the Watch to be ready. Used to be much faster. Also, with it connected via Bluetooth (right?) it seems it should be almost instant when I'm near the car.

**Comment by ttwajsb** [comment]
Now I've got my fingers crossed that it won't need to be authenticated so often.

**Comment by ttwajsb** [comment]
Remote Start very inconsistent. Thankfully not routinely having to reauthenticate.

**Comment by ttwajsb** [comment]
Not true...![image](https://github.com/itsMeDavidV/Watchla-Support/assets/9358980/090ff5ab-fbc2-4ebb-9bf0-3a0aa91bc4b3)

**Comment by ttwajsb** [comment]
Alas, I'm about ready to give up and wait till Tesla makes a native Watch app. I appreciate your effort to make this work, David. It was delightful during the months or more it was letting me remote start my car. Loved that, and sad having to live without it now.

**Comment by ttwajsb** [comment]
Was excited to get and try the latest Watchla 3.7.5. 
As recommended along the way, logged out of iPhone app (had to force the logout).
Went thru the new two stage process to authenticate first the iPhone then the Watch. That flowed perfectly. 
Alas, Remote Start still doesn't work. Multiple attempts. 
Also, getting this 403 error when trying to fart. Tried numerous times. Initially it immediately went to this screen. Just now, 13 hours later, the fart icon went into a long spinning mode before giving this error. 
![image](https://github.com/itsMeDavidV/Watchla-Support/assets/9358980/57b835e6-d587-44b5-bc24-8788bb274588)

**Comment by itsMeDavidV** [comment]
Hi @ttwajsb sorry about the inconvenience this transition has caused you. Adapting to Tesla's new API and a new watch app at the same time has been quite the undertaking, but rest-assured we're diligently working to iron out the remaining kinks.

Regarding the 403 error you're seeing, Tesla only gives these out for 2 reasons:

1: you haven't authorized Watchla access to a certain scope (eg: vehicle commands) (this can be checked in your Tesla account)
2: or because the client (app) attempted to send an unsigned vehicle command for a command that is required to be signed.

Remote Start should work as Watchla is adapted to send signed Remote Start commands however there could be an error down the line. Can you confirm where  in Watchla you triggered Remote Start from? Additionally I'd recommend ensuring that you have enabled Watchla to have the proper permissions (as mentioned above).

On the contrary for the Fart command, unfortunately Tesla has not provided us developers with a way to sign Fart commands and have for the time being allowed unsigned Fart commands. So in your case the only likely reason you would see a 403 error for the Fart command is if you have not allowed the necessary permission for Watchla (again, I double check this as mentioned above).

Are you seeing this error only for those commands? Or are commands in general not working for you?

**Comment by ttwajsb** [comment]
Funk, funk, funk. 
Remote Start and Fart still not working. Yet from my wrist I can toggle Climate, lock-unlock, open frunk, etc. 

Have completely signed out in Tesla and from app, removed from Watch, deleted app (including its data).

Then started from scratch, carefully following all instructions. This included separately authorizing both iPhone and Watch via the Tesla login. 

app version: 3.8.1 (446)

iPhone 14 Pro with iOS 17.5.1
Watch Series 4 cellular (NTUW2LL/A) with watchOS 10.4 (21T216)

2018 Tesla Model 3 running v11.1 (current as of today, plus FSD v12.3.6) 

David, I was LOVING this app when it worked. So sad to not be able to use it for Remote Start, which was my favorite feature by far. 

I regret that I can no longer recommend it to my friends and in reviews. :(

PS - As seen in this photo, the Remote Start function itself is actually interacting with the car, but seemingly only in receive mode, not send mode. That is, sending the instruction to start doesn't work yet the Watch can receive within the Remote Start interface the fact that the car is now on Drive. 
![image](https://github.com/itsMeDavidV/Watchla-Support/assets/9358980/3d7a545f-7fe9-4ece-9abc-ccca7a2d2ec3)
